packages/prompts/src/index.ts

```diff
@@ -374,14 +375,140 @@ export const multiselect = <Options extends Option<Value>[], Value extends Primi
+		} else if (state === 'group-active') {
+			return `${prefix}${color.cyan(S_CHECKBOX_ACTIVE)} ${color.dim(label)}`;
+		} else if (state === 'group-active-selected') {
+			return `${prefix}${color.green(S_CHECKBOX_SELECTED)} ${color.dim(label)}`;
+		} else if (state === 'selected') {
+			return `${color.dim(prefix)}${color.green(S_CHECKBOX_SELECTED)} ${color.dim(label)}`;
+		} else if (state === 'cancelled') {
+			return `${color.strikethrough(color.dim(label))}`;
+		} else if (state === 'active-selected') {
+			return `${color.dim(prefix)}${color.green(S_CHECKBOX_SELECTED)} ${label} ${
+				option.hint ? color.dim(`(${option.hint})`) : ''
+			}`;
+		} else if (state === 'submitted') {
+			return `${color.dim(label)}`;
+		}
+		return `${color.dim(prefix)}${color.dim(S_CHECKBOX_INACTIVE)} ${color.dim(label)}`;
+	};
+
+	return new GroupMultiSelectPrompt({
+		options: opts.options,
+		initialValues: opts.initialValues,
+		required: opts.required ?? true,
+		cursorAt: opts.cursorAt,
+		validate(selected: Value[]) {
+			if (this.required && selected.length === 0)
+				return `Please select at least one option.\n${color.reset(
+					color.dim(
+						`Press ${color.gray(color.bgWhite(color.inverse(' space ')))} to select, ${color.gray(
+							color.bgWhite(color.inverse(' enter '))
+						)} to submit`
+					)
+				)}`;
+		},
+		render() {
+			let title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+
+			switch (this.state) {
+				case 'submit': {
+					return `${title}${color.gray(S_BAR)}  ${this.options
+						.filter(({ value }) => this.value.includes(value))
+						.map((option) => opt(option, 'submitted'))
+						.join(color.dim(', '))}`;
+				}
+				case 'cancel': {
+					const label = this.options
+						.filter(({ value }) => this.value.includes(value))
+						.map((option) => opt(option, 'cancelled'))
+						.join(color.dim(', '));
+					return `${title}${color.gray(S_BAR)}  ${
+						label.trim() ? `${label}\n${color.gray(S_BAR)}` : ''
+					}`;
+				}
+				case 'error': {
+					const footer = this.error
+						.split('\n')
+						.map((ln, i) =>
+							i === 0 ? `${color.yellow(S_BAR_END)}  ${color.yellow(ln)}` : `   ${ln}`
+						)
+						.join('\n');
+					return `${title}${color.yellow(S_BAR)}  ${this.options
+						.map((option, i, options) => {
+							const selected = this.value.includes(option.value) || (option.group === true && this.isGroupSelected(option.value));
+							const active = i === this.cursor;
+							const groupActive = !active &&  typeof option.group === 'string' && this.options[this.cursor].value === option.group;
+							if (groupActive) {
+								return opt(option, selected ? 'group-active-selected' : 'group-active', options);
+							}
+							if (active && selected) {
+								return opt(option, 'active-selected', options);
+							}
+							if (selected) {
+								return opt(option, 'selected', options);
+							}
+							return opt(option, active ? 'active' : 'inactive', options);
```

Comment:
This seems like logic that should be moved into `@clack/core` somehow...

Reply:
Yeah I don't know what the API should look like, but the fact that we have to track state in the rendering logic feels leaky. That's the whole point of `@clack/core`!